pkg/operator/controller/ingress/deployment.go

```diff
@@ -311,6 +311,27 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		}
 	}
 
+	// Configure topology constraints to spread replicas across availability
+	// zones.  We want to allow scheduling more replicas than there are AZs,
+	// so we specify "ScheduleAnyway".  We want to allow scheduling a
+	// newer-generation replica on the same node as an older-generation
+	// replica where the deployment strategy allows an depends on doing so,
+	// so we specify a label selector with the deployment's hash.
+	deployment.Spec.Template.Spec.TopologySpreadConstraints = []corev1.TopologySpreadConstraint{{
+		MaxSkew:           int32(1),
+		TopologyKey:       "topology.kubernetes.io/zone",
```

Comment:
```suggestion
		TopologyKey:       corev1.LabelTopologyZone,
```

Might as well use that Topology key from corev1, right?

Reply:
Done.  Thanks!